tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java

```diff
@@ -1821,6 +1821,18 @@ public static void test_schema_reflection() throws Exception {
 
 		conn.close();
 	}
+	
+	public static void test_get_tables_with_catalog() throws Exception {
+		Connection conn = DriverManager.getConnection("jdbc:duckdb:");
+		String catalog = conn.getCatalog();
+		DatabaseMetaData databaseMetaData = conn.getMetaData();
```

Comment:
Could we add some tests that verify the `getTables` correctly takes the catalog into account, by e.g. attaching an in-memory database and creating tables there?

Reply:
Sure! Will pick it up tonight.

Reply:
@Mytherin - something like this? https://github.com/duckdb/duckdb/pull/6230/commits/df0303c652e2b874ec5005b7faf712e80f47dee4